craftbook-bukkit/src/main/java/org/enginehub/craftbook/mechanics/area/Gate.java

```diff
@@ -374,27 +365,46 @@ public Block getBlockBase(Block sign) throws InvalidMechanismException {
         Block gateBlock = null;
 
         if (sign != null) {
-            int x = sign.getX();
-            int y = sign.getY();
-            int z = sign.getZ();
             Material expectedType = getStoredType((Sign) sign.getState(false));
 
-            for (int x1 = x - searchRadius; x1 <= x + searchRadius; x1++) {
-                for (int y1 = y - searchRadius; y1 <= y + searchRadius * 2; y1++) {
-                    for (int z1 = z - searchRadius; z1 <= z + searchRadius; z1++) {
-                        Block offsetBlock = sign.getWorld().getBlockAt(x1, y1, z1);
-                        BlockData offsetBlockData = offsetBlock.getBlockData();
-                        if (expectedType != null) {
-                            // If we have an expected type set, require that one to be used.
-                            if (offsetBlockData.getMaterial() == expectedType) {
-                                gateBlock = offsetBlock;
-                                break;
-                            }
-                        } else if (Blocks.containsFuzzy(blocks, BukkitAdapter.adapt(offsetBlockData))) {
-                            gateBlock = offsetBlock;
-                            break;
+            Deque<Block> enumerationQueue = new ArrayDeque<>();
+            Set<Block> visited = new HashSet<>();
+
+            enumerationQueue.add(sign);
+            visited.add(sign);
+
+            enumerationLoop: while (!enumerationQueue.isEmpty()) {
+                Block currentOrigin = enumerationQueue.poll();
+
+                if (Math.abs(currentOrigin.getX() - sign.getX()) >= searchRadius)
```

Comment:
Could you please ensure that the if statements/etc here keep the braces and follow the style guide?